app/javascript/packs/commentDropdowns.js

```diff
@@ -15,51 +15,99 @@ const handleCopyPermalink = (closeDropdown) => {
+      });
 
-    // Add actual link location (SEO doesn't like these "useless" links, so adding in here instead of in HTML)
-    const dropdownElement = document.getElementById(dropdownContentId);
-    const reportAbuseWrapper = dropdownElement.querySelector(
-      '.report-abuse-link-wrapper',
-    );
-    if (reportAbuseWrapper) {
-      reportAbuseWrapper.innerHTML = `<a href="${reportAbuseWrapper.dataset.path}" class="crayons-link crayons-link--block">Report abuse</a>`;
-    }
+      // Add actual link location (SEO doesn't like these "useless" links, so adding in here instead of in HTML)
+      const reportAbuseWrapper = dropdownElement.querySelector(
+        '.report-abuse-link-wrapper',
+      );
+      if (reportAbuseWrapper) {
+        reportAbuseWrapper.innerHTML = `<a href="${reportAbuseWrapper.dataset.path}" class="crayons-link crayons-link--block">Report abuse</a>`;
+      }
 
-    // Initialize the "Copy link" functionality
-    dropdownElement
-      .querySelector('.permalink-copybtn')
-      ?.addEventListener('click', handleCopyPermalink(closeDropdown));
+      // Initialize the "Copy link" functionality
+      dropdownElement
+        .querySelector('.permalink-copybtn')
+        ?.addEventListener('click', handleCopyPermalink(closeDropdown));
 
-    dropdownTrigger.dataset.initialized = 'true';
+      dropdownTrigger.dataset.initialized = 'true';
+    }
   }
 };
 
+/**
+ * Function to retrieve the Profile Preview Card for newly added comments, and asynchronously replace the placeholder
+ *
+ * @param {HTMLElement} placeholderElement The <span> placeholder element to be replaced by the preview dropdown
+ */
+const fetchMissingProfilePreviewCard = async (placeholderElement) => {
+  const htmlContent = await window
+    .fetch(
+      `/profile_preview_card/show?userid=${placeholderElement.dataset.jsCommentUserId}&preview_card_id=${placeholderElement.dataset.jsDropdownContentId}`,
+    )
+    .then((res) => res.text());
+
+  const generatedElement = document.createElement('div');
+  generatedElement.innerHTML = htmlContent;
+
+  placeholderElement.parentNode.replaceChild(
+    generatedElement.firstElementChild,
+    placeholderElement,
+  );
+
+  // Make sure the button inside the dropdown is initialized
+  initializeAllFollowButts();
+};
+
+/**
+ * When a new comment is added to a dicussion, the preview card dropdown must be dynamically fetched.
+ * This function detects if a preview card placeholder has been added in a given mutation and initiates the profile card fetch.
+ **/
+const checkMutationForProfilePreviewCardPlaceholder = (mutation) => {
+  mutation.addedNodes.forEach((node) => {
+    if (node.nodeType === Node.ELEMENT_NODE) {
+      const placeholder = node.getElementsByClassName(
+        'preview-card-placeholder',
+      )[0];
+      if (placeholder) {
+        fetchMissingProfilePreviewCard(placeholder);
+      }
+    }
+  });
+};
+
 const observer = new MutationObserver((mutationsList) => {
```

Comment:
Nice! 👏 I knew these existed but I've never used one and it's cool to see this in action.

Reply:
We took this approach for the Gist liquid tags in another PR as well. https://github.com/forem/forem/pull/13407#pullrequestreview-646200476 😎